src/robot/libdocpkg/datatypes.py

```diff
@@ -58,9 +58,11 @@ def for_type(cls, type, converters):
         elif not converter.type:
             return cls(cls.CUSTOM, converter.type_name, converter.doc,
                        converter.value_types)
-        else:
+        elif converter.type in STANDARD_TYPE_DOCS:
             return cls(cls.STANDARD, converter.type_name,
                        STANDARD_TYPE_DOCS[converter.type], converter.value_types)
+        else:
+            return None
```

Comment:
Why are these changes needed?

Reply:
Due to a defect in the existing code. While testing I ran into a KeyError in what is now line 63. The code assumed that if there is a converter and it has a type, that it must be in STANDARD_TYPE_DOCS. At one point it wasn't.

To be honest, I am not sure anymore if this was only to improve error reporting during work in progress, or that there are still active paths in the running code that reach this point. Anyway, the KeyError was not very helpful, so I kept the improvement regardless.

Reply:
If it's a recognized converted, Libdoc should have docs for it and `return None` is thus risky. I agree `KeyError` isn't good either convenient (that occurred recently after I added `pathlib.Path` converter and forgot to add docs), but it would be better to use something like `raise ValueError(f'No documentation for {type(converter).__nama__}.')` instead of `return None´.

Reply:
I would have to reproduce the issue to really remember what was going on. If I recall correctly it had something to do with new composed types being created at runtime from existing types, for which Libdoc didn't find the docs. Returning None had in that case identical behavior as when there isn't any converter (line 57). I think it still showed the name, but no link to the type's details.

Reply:
In that case it would probably be best to replace `return None` with some meaningful exception. Most likely it occurs only if there's a bug in our code (all recognized converters should have docs!) and a good exception makes understanding the situation better. If there are in fact legitimate cases where a converter shouldn't have docs, then this code is fine but there should be a unit test validating it.

Reply:
I am clearly missing something obvious. This very basic example already fails to show the type link. It also doesn't raise the exception though.

`attempt.py`
```
class MyStuff:
    """bla bla"""

class attempt:
    def dummy(self, x: MyStuff):
        pass
```

It shows Keyword dummy with argument `x <MyStuff>`, but it does not link to show '_bla bla_'

Reply:
Libdoc doesn't work like this. It shows type info only for types it converts. That means std types like `int` and custom types that are separately registered.

Reply:
The real issue here seems to be that I was using the same construct as inside Robot in an external library. So, using `@TypeConverter.register` instead of `ROBOT_LIBRARY_CONVERTERS=` or `@library(converters=`. Functionally it works, but it somehow has a side effect when using Libdoc. On the other hand, I am unable to reach the nested_type info when using the custom converter approach.